agent/grpc-internal/resolver/resolver.go

```diff
@@ -176,14 +213,35 @@ func (s *ServerResolverBuilder) RemoveServer(areaID types.AreaID, server *metada
 	}
 }
 
+func (s *ServerResolverBuilder) shouldIgnoreServer(areaID types.AreaID, server *metadata.Server) bool {
```

Comment:
can we add some context around why these conditions should ignore server? What are we filtering out?